nep-12.mediawiki

```diff
@@ -0,0 +1,2130 @@
+                "value": "00e1f505"
+              }
+            ]
+          }
+        },
+        {
+          "contract": "0x849d095d07950b9e56d0c895ec48ec5100cfdff1",
+          "state": {
+            "type": "Array",
+            "value": [
+              {
+                "type": "ByteArray",
+                "value": "7472616e73666572"
+              },
+              {
+                "type": "ByteArray",
+                "value": "296ac124021a71c449a9bad320c16429b08ad6ee"
+              },
+              {
+                "type": "ByteArray",
+                "value": "b1fdddf658ce5ff9f83e66ede2f333ecfcc0463e"
+              },
+              {
+                "type": "ByteArray",
+                "value": "00e1f505"
+              }
+            ]
+          }
+        }
+      ]
+    }
+  ]
+}
+</pre>
+
+
+===Write Methods===
+====send====
+Invoke a transfer of a specified amount of a given asset from the connected account to another account.
+
+
+'''Method Interface'''
+
+<pre>
+function send({
+  fromAddress: string,
+  toAddress: string,
+  asset: string,
+  amount: string,
+  remark?: string,
+  fee?: string,
+  network?: string,
+  broadcastOverride?: boolean,
+}): Promise<SendOutput>
+</pre>
+
+
+'''Input arguments'''
+
+<pre>
+interface SendArgs {
+  fromAddress: string; // Address of the connected account to send the assets from
+  toAddress: string; // Address of the receiver of the assets to be sent
+  asset: string; // Asset script hash to be sent
+  amount: string; // The parsed amount of the asset to be sent
+  remark?: string; // (Optional) Description of the transaction to be made
+  fee?: string; // (Optional) The parsed amount of network fee (in GAS) to include with transaction
+  network?: string; //  Network to submit this request to. If omitted, will default to network the wallet is currently set to.
+  broadcastOverride?: boolean; // In the case that the dApp would like to be responsible for broadcasting the signed transaction rather than the wallet provider
+}
+</pre>
+
+
+'''Success return value'''
+
+<pre>
+interface SendOutput {
+  txid: string; // The transaction ID of the send invocation
+  nodeUrl?: string; // The node which the transaction was broadcast to. Returned if transaction is broadcast by wallet provider
+  signedTx?: string; // The serialized signed transaction. Only returned if the broadcastOverride input argument was set to True
+}
+</pre>
+
+
+'''Error return value'''
+
+<pre>
+interface Error {
+  type: string; // `NO_PROVIDER`|`CONNECTION_REFUSED`|`SEND_ERROR`|`MALFORMED_INPUT`|`CANCELED`|`INSUFFICIENT_FUNDS`
+  description: string;
+  data: string;
+}
+</pre>
+
+
+'''Example'''
+
+<pre>
+send({
+  fromAddress: 'ATaWxfUAiBcQixNPq6TvKHEHPQum9bx79d',
+  toAddress: 'ATaWxfUAiBcQixNPq6TvKHEHPQum9bx79d',
+  asset: '602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7',
+  amount: '0.0001',
+  remark: 'Hash puppy clothing purchase. Invoice#abc123',
```

Comment:
`remark` usually has options too... perhaps add some `remarkType` option? Or just let Provider choose one?

Reply:
The value in the `remark` field is attached to the transaction as a transaction attribute. What did you have in mind for `remarkType`?

Reply:
I think there are many types of remark, on transactionattributetype class. few are remarks, few are hashes. Remark0 is often used,but it could be others too.